![Callysto.ca Banner](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-top.jpg?raw=true)

# Elementary Climate Change

A number of indicators in the environment around use can tell us about the extent of climate change in the world today.
* Duration and Thickness of Sea Ice
* Ocean Levels
* Permafrost Changes
* Number of Forest Fires

In this notebok we will explore data on each of these indicators. At the end you will be asked to reflect on some questions to determine what you have learned based on the data. In order to examine this topic we took. data from the [Global Climte Change-Vital Signs of the Planet](https://climate.nasa.gov/) website published by NASA. This website has many other interesting details on it including other visualizations, facts and some information on how we can mitigate climate change. Parts of this website will be interesting to teachers and parts will also be useful for students to explore.

Run the code cell below to import the libraries we need for this project. Libraries are pre-made code that make it easier to analyze our data. 

Pandas is a library that helps us with data analysis, and plotly express is a library that helps us to make visualizations. 

Without importing these libraries we would have to use much more code to analyze our data and generate visualizations. We import the libraries with abbreviations, or aliases, so that we have less typing to do in each line of our code below. The first two lines in the cell below import libraries. The cells below that import our data sets so that we can work with them to build visualizations. 

In [1]:
import pandas as pd
import plotly.express as px
ice_sheets = pd.read_csv('https://raw.githubusercontent.com/callysto/data-files/main/Science/ElementaryClimateChange/icesheets.csv')
ocean_warming = pd.read_csv('https://raw.githubusercontent.com/callysto/data-files/main/Science/ElementaryClimateChange/oceanwarm.csv')
temperature = pd.read_csv('https://raw.githubusercontent.com/callysto/data-files/main/Science/ElementaryClimateChange/temperature.csv')

## Sea Ice

The code block below build a data frame from the data on sea ice. A data frame is a way that the python programming language we use stores data. A data frame is kind of like a spreadsheet.


In [2]:
ice_sheets = pd.read_csv('https://raw.githubusercontent.com/callysto/data-files/main/Science/ElementaryClimateChange/icesheets.csv')
ice_sheets

Time  Antarctic Mass  Antartic mass 1-sigma uncertainty
0    2002.29            0.00                             178.90
1    2002.35           17.63                             102.47
2    2002.62          -62.83                              83.09
3    2002.71           43.30                              97.19
4    2002.79           59.90                              61.82
..       ...             ...                                ...
213  2022.79         2415.28                              33.54
214  2022.87         2349.74                              39.99
215  2022.96         2352.28                              47.38
216  2023.04         2425.63                              55.48
217  2023.12         2481.37                              65.02

[218 rows x 3 columns]

What do you notice about the data frame we created with that code? 

From looking at the numbers at the data frame we can see it has 3 columns and 218 rows. Each row represents a data point. If you look at the last row in the data frame above you can see it is labelled 217 but the numbers below the data set say that we have 218 rows. That may seem a bit strange at first but that is because python starts counting at 0.

We can see the first 5 rows and the last 5 rows of data in the data frame above. If we want to see the first twenty rows of data we can run the code cell with the instruction below. 

In [10]:
ice_sheets.head(20)

Time  Antarctic Mass  Antartic mass 1-sigma uncertainty
0   2002.29            0.00                             178.90
1   2002.35           17.63                             102.47
2   2002.62          -62.83                              83.09
3   2002.71           43.30                              97.19
4   2002.79           59.90                              61.82
5   2002.87          -71.58                              58.68
6   2002.96          -48.89                              59.19
7   2003.04          -48.71                              64.73
8   2003.12         -200.79                              48.10
9   2003.20         -171.01                              42.95
10  2003.29          -44.00                              43.58
11  2003.36            1.53                              48.24
12  2003.54         -130.07                              37.86
13  2003.62         -123.81                              39.87
14  2003.71         -133.11                              39.61
15  2003.79          -50.21                              40.19
16  2003.87         -110.16                              40.89
17  2003.96         -275.30                              39.82
18  2004.02         -331.67                              43.72
19  2004.13         -490.94                              40.46

If we want to see the last twenty rows in the data set we can run the code with the instruction below. 

In [11]:
ice_sheets.tail(20)

Time  Antarctic Mass  Antartic mass 1-sigma uncertainty
198  2021.54         2675.54                              26.69
199  2021.62         2615.17                              26.73
200  2021.71         2702.15                              26.69
201  2021.79         2504.48                              26.69
202  2021.87         2743.28                              26.78
203  2021.96         2646.60                              26.71
204  2022.04         2740.35                              26.73
205  2022.12         2699.54                              26.86
206  2022.20         2623.45                              27.50
207  2022.29         2384.55                              28.26
208  2022.37         2289.38                              29.08
209  2022.45         2310.88                              29.87
210  2022.54         2419.74                              30.71
211  2022.62         2374.28                              31.80
212  2022.71         2359.21                              32.68
213  2022.79         2415.28                              33.54
214  2022.87         2349.74                              39.99
215  2022.96         2352.28                              47.38
216  2023.04         2425.63                              55.48
217  2023.12         2481.37                              65.02

You can view any number of rows starting from the beginning of the data frame by changing the number in the brackets in the cell with the word head.

You can also change the number in the brackets in the cell with the word tail to view a different number of rows at the bottom of the data frame. 

Run the code below to generate a line graph about the ice sheets. 

In [14]:
px.line(ice_sheets, x='Time', y='Antarctic Mass', title='Mass of Ice Sheets Over Time')

We can see that the mass of the ice sheets in the Antarctic are fluctuating. Why do you think this is? What does this tell you about climate change?
* What kind of a research question could you pose to find an answer to this? 
* Where do you think you might find an answer to this question? What kind of sources should you look at?

## Permafrost Changes

## Ocean Level

In [4]:
sea_level = pd.read_csv('https://raw.githubusercontent.com/callysto/data-files/main/Science/ElementaryClimateChange/sealevel.csv')
sea_level

Altimeter Type  Merged File Cycle Number  \
0                  0                        11   
1                  0                        12   
2                  0                        13   
3                  0                        14   
4                  0                        15   
...              ...                       ...   
1106               0                       118   
1107               0                       119   
1108               0                       120   
1109               0                       121   
1110               0                       122   

      Year + Fraction of Year (Mid-Cycle)  Observations  \
0                             1993.011526        452600   
1                             1993.038692        447591   
2                             1993.065858        459462   
3                             1993.093025        410064   
4                             1993.120191        446813   
...                                   ...           ...   
1106                          2023.065279        467851   
1107                          2023.092445        467307   
1108                          2023.119611        427658   
1109                          2023.146777        470397   
1110                          2023.173943        469810   

      Weighted Observations  GMSL - Gia not Applied  \
0                 327401.31                  -38.61   
1                 324498.41                  -42.01   
2                 333018.19                  -41.97   
3                 297483.19                  -42.71   
4                 321635.81                  -37.88   
...                     ...                     ...   
1106              337933.19                   58.72   
1107              338525.91                   58.05   
1108              309056.59                   58.29   
1109              339192.31                   57.34   
1110              338049.50                   57.21   

      Std Dev of GMSL - Gia not Applied  Smoothed GMSL - Gia not Applied  \
0                                 89.88                           -38.78   
1                                 90.93                           -39.80   
2                                 87.31                           -39.65   
3                                 90.79                           -39.69   
4                                 90.29                           -38.78   
...                                 ...                              ...   
1106                              87.03                            58.91   
1107                              86.63                            58.59   
1108                              89.01                            58.38   
1109                              90.25                            58.26   
1110                              88.61                            58.19   

      GMSL - Gia Applied  STd Dev of GMSL - Gia Applied  \
0                 -38.61                          89.88   
1                 -42.00                          90.93   
2                 -41.96                          87.31   
3                 -42.69                          90.79   
4                 -37.85                          90.29   
...                  ...                            ...   
1106               66.03                          87.15   
1107               65.34                          86.65   
1108               65.59                          88.63   
1109               64.67                          89.81   
1110               64.53                          88.39   

      Smoothed GMSL - Gia Applied  Smoothed GMSL - Gia Applied Signal Removed  \
0                          -38.78                                      -38.56   
1                          -39.80                                      -39.09   
2                          -39.64                                      -38.56   
3                          -39.67                                      -38.32   
4       

In [7]:
px.line(sea_level, x='Year + Fraction of Year (Mid-Cycle)', y='GMSL - Gia Applied', title='Ocean Level Over Time')

We can see that the general trend in the data is that ocean level is rising. Why do you think this is? How do you think that relates to climate chnage? Read on to find out more about how climate change is related to sea level. 

**Sea Level and Climate Change**

According to the [NASA Website](https://climate.nasa.gov/vital-signs/sea-level) that published our data sea levels and climte change are related in two ways. You can read more about it by clicking on the hyperlink.



## Forest Fires

## Extreme Weather



## Questions to Consider


* Which data did you find most interesting? Why?
* What does this data tell you about climate change? How can you support your answer?
* What actions can you take to help slow down climate change?

How can you share what you have learned with your classmates and those around you? Can you make a poster, powerpoint presentation or give a speech? What other methods could you consider?

## More Information To Explore

Callysto has created other resources using data that is related to topics explored in this notebook. For an extension to this notebook you may want to check out the resources linked below. 

 Check out the notebook Callysto created on the costliest disasters for an interesting [look at extreme weather events in recent history.](https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2Fcallysto%2Fdata-viz-of-the-week&branch=main&subPath=costliest-disasters/costliest-natural-disasters.ipynb&depth=1)

As the climate has been changing the level of sea ice has been changing in the world. Callysto created a data visualization on [changes in sea ice.](https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2Fcallysto%2Fdata-viz-of-the-week&branch=main&subPath=ice-coverage-best-fit/sea-ice.ipynb&depth=1) 

You may also be interested in looking at our recent data visualizaton on the [warmest January temperatures.](https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2Fcallysto%2Fdata-viz-of-the-week&branch=main&subPath=january-temperatures/january-temperatures.ipynb&depth=1)


[![Callysto.ca License](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-bottom.jpg?raw=true)](https://github.com/callysto/curriculum-notebooks/blob/master/LICENSE.md)